### ISSUE TITLE
Detect input format based on file name extension

### ISSUE THREAD

**Issue by ryenus** [mikefarah/yq#1440]
**Please describe your feature request.**

Is it possible for yq to be a little more smart to detect the input file type based on the file extension? At least for common ones that yq already/would support, like:
- yml, yaml
- json
- xml
- csv
- properties

**Describe the solution you'd like**
When an xml file (`*.xml`) is used as input, it should be obvious for yq to treat the input as xml, rather than yaml.

```xml
<p>
  <a>1</a> <!-- comment: message -->
</p>
```

And we run a command:
```bash
yq '... comments="" | .p' data1.xml
```

it could output
```yaml
a: "1"
```

**Describe alternatives you've considered**
it works by always explicitly specify the input file type, e.g.:
```bash
yq -px '... comments="" | .p' data1.xml
# a: "1"
```

**Additional context**
The request is related to https://github.com/mikefarah/yq/issues/1438#issuecomment-1319942685

**Comment by mikefarah** [comment]
Neat idea 👍🏼  - happy to take PRs - otherwise I'll get to it eventually

**Comment by mikefarah** [comment]
Released in 4.32.1 - Thanks again @ryenus   :)  I ended up taking your solution a little further, the output format is also automatically set when the input format is not specified :)

**Comment by ryenus** [comment]
> Released in 4.32.1 - Thanks again @ryenus :) I ended up taking your solution a little further, the output format is also automatically set when the input format is not specified :)

@mikefarah, I'm really happy this PR has made it, and having it taken further is more than I would have expected, cheers! 🎉🎉🎉
